mw-webapp/src/component/image/Image.tsx

```diff
@@ -25,21 +32,72 @@ interface ImageProps {
    * Data attribute for cypress testing
    */
   dataCy?: string;
+
+  /**
+   * Image respond to clicks
+   */
+  isZoomable?: boolean;
+
+  /**
+   * Large Image
+   */
+  isZoomed?: boolean;
+
+  /**
+   * Control Image outside
+   */
+  onOpenChange?: (arg: boolean) => void;
 }
 
 /**
- * Сomponent for displaying images
+ * Component for displaying images
  */
 export const Image = (props: ImageProps) => {
-  const className = clsx(styles.image, props.className);
+  const imageClass = clsx(styles.image, props.className);
+  const [isOpen, setIsOpen] = useState(false);
+
+  /**
+   * Toggle modal state
+   */
+  const handleOpenChange = (value: boolean) => {
+    setIsOpen(value);
+  };
 
-  return (
+  const imageElement = (
     <img
       src={props.src}
       alt={props.alt}
-      className={className}
+      className={imageClass}
       data-cy={props.dataCy}
     />
   );
-};
 
+  return props.isZoomable
+    ? (
+      <Modal
+        trigger={
+          <img
+            src={props.src}
+            alt={props.alt}
+            className={imageClass}
+            data-cy={props.dataCy}
+            onClick={() => handleOpenChange(true)}
+          />
+        }
+        content={
+          <img
+            src={props.src}
+            alt={props.alt}
+            className={clsx(styles.zoomedImage, props.className)}
+            onClick={() => handleOpenChange(false)}
+          />
+        }
+        isOpen={props.isZoomed ?? isOpen}
+        className={imageClass}
+        cy={props.cy}
+      />
+    )
+    : (
+      imageElement
+    );
+};
```

Comment:
you have 3 img with 2 cy attributes. It looks strange a little, think about it how to test it properly. I think only ine cy id required for our case.

Good. Lets add tests for new functionality and check that everything works in storybook and in the demo